lib/config.js

```diff
@@ -457,6 +464,7 @@ AWS.Config = AWS.util.inherit({
     sslEnabled: true,
     s3ForcePathStyle: false,
     s3BucketEndpoint: false,
+    s3DisableBodySigning: true,
```

Comment:
`sslEnabled` is also defaulted to `true`. The user may have `sslEnabled` set globally or at the client level, and depending on the order that properties get merged in (which is not guaranteed when using JavaScript objects), we could incorrectly set the 'default' `s3DisableBodySigning`. For example, if the user globally set `sslEnabled` to false, and set it to true on the s3 client, if the `s3DisableBodySigning` property was merged before `sslEnabled`, it would default to `false` incorrectly. I think this complicates the behavior unnecessarily, and could lead to trickier behavior to troubleshoot.

I can update the documentation to specify that this config is also only applied when using https. Besides `sslEnabled`, users can also specify their own endpoint that uses something other than `https`, so it'd be good to document this config for that scenario as well.

I am hesitant to throw an error if both settings are explicitly set. I think documentation is the better route to go, since we can reliably handle the case where `sslEnabled: false` and `s3DisableBodySigning: true`, and the behavior is identical to what we're currently doing.